python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/api/API_CONVENTIONS.md

```diff
@@ -0,0 +1,302 @@
+# Dagster Plus API CLI Conventions
+
+This document outlines the conventions for implementing CLI commands in the `dg api` namespace, following GitHub CLI best practices.
+
+## Command Structure
+
+The API commands follow the pattern: `dg api <noun> <verb>`
+
+### Examples:
+
+```bash
+dg api deployment list
+dg api deployment view <id>
+dg api secret list
+dg api secret create
+dg api agent list
+```
+
+This mirrors GitHub CLI patterns like:
+
+- `gh repo list`
+- `gh issue create`
+- `gh pr view`
+
+## File Organization
+
+### Directory Structure:
+
+```
+dagster_dg_cli/cli/plus/api/
+├── __init__.py                 # Main API group
+├── deployment.py               # Deployment noun with all verbs
+├── secret.py                   # Secret noun with all verbs
+├── agent.py                    # Agent noun with all verbs
+├── check.py                    # Asset check noun with all verbs
+├── code_location.py            # Code location noun with all verbs
+├── env_var.py                  # Environment variable noun with all verbs
+└── shared.py                   # Shared utilities
+```
+
+### Implementation Pattern:
+
+Each noun file (e.g., `deployment.py`) contains:
+
+1. **Noun group** - Click group for the resource
+2. **Verb commands** - Individual commands for operations
+3. **Helper functions** - Shared utilities for that noun
+
+## Standard Flags
+
+### Required for ALL API Commands:
+
+- `--json` flag for machine-readable JSON output
+- Human-readable table format as default
+
+### Common Optional Flags:
+
+- `--limit <n>` for pagination
```

Comment:
this may be an opportunity to move to a limit/pagination by default setup whenever possible, instead of having to shut the barn door later